modules/engines/engine-odata/src/main/java/org/eclipse/dirigible/engine/odata2/transformers/DBMetadataUtil.java

```diff
@@ -249,27 +249,4 @@ public String getArtifactSchema(String artifactName, String[] types) throws SQLE
             return null;
         }
     }
-
-    public HashMap<String, String> getSynonymTargetObjectMetadata(String synonymName) throws SQLException {
-        HashMap<String, String> targetObjectMetadata = new HashMap<>();
-
-        try (Connection connection = dataSource.getConnection()) {
-            PreparedStatement prepareStatement = connection.prepareStatement("SELECT * FROM \"SYS\".\"SYNONYMS\" WHERE \"SYNONYM_NAME\" = ?");
-            prepareStatement.setString(1, synonymName);
-
-            try (ResultSet resultSet = prepareStatement.executeQuery()) {
-                if (resultSet.next()) {
-                    String name = resultSet.getString("OBJECT_NAME");
-                    String schema = resultSet.getString("OBJECT_SCHEMA");
-                    String type = resultSet.getString("OBJECT_TYPE");
-
-                    targetObjectMetadata.put(ISqlKeywords.KEYWORD_TABLE, name);
-                    targetObjectMetadata.put(ISqlKeywords.KEYWORD_SCHEMA, schema);
-                    targetObjectMetadata.put(ISqlKeywords.KEYWORD_TABLE_TYPE, type);
-                }
-            }
-        }
-
-        return targetObjectMetadata;
-    }
```

Comment:
@MarinHadzhiev, @delchev, @BorisNen Should we merge it as proposed in the PR, or should we wait for this to be implemented as part of the HANA dialect?

Reply:
@ThuF We have a meeting Monday morning to discuss the implementation option with SQL dialect. Based on the effort estimation will decide whether to merge this one or rework it. Let's keep it open for now.